File: src/0201-0300/228 - Summary Ranges/summary_ranges.go
package main

import (
	"fmt"
)

func summaryRanges(nums []int) []string {
	list := []string{}
	n := len(nums)

	for i := 0; i < n; i++ {
		start := nums[i]
		println(i)
		for i+1 < n && nums[i+1] == nums[i]+1 {
			i++
		}
		if start == nums[i] {
			list = append(list, fmt.Sprint(start))
		} else {
			list = append(list, fmt.Sprint(start, "->", nums[i]))
		}
	}

	return list
}

File: src/0201-0300/228 - Summary Ranges/summary_ranges.rs
impl Solution {
    pub fn summary_ranges(nums: Vec<i32>) -> Vec<String> {
        let mut list = Vec::new();
        let n = nums.len();
        let mut i = 0;

        while i < n {
            let start = nums[i];

            while i + 1 < n && nums[i + 1] == nums[i] + 1 {
                i += 1;
            }

            if start == nums[i] {
                list.push(format!("{}", start));
            } else {
                list.push(format!("{}->{}", start, nums[i]));
            }

            i += 1;
        }

        list
    }
}
